seaworthy/tests-core/test_logs.py

```diff
@@ -179,8 +179,12 @@ def attach_socket(self, params):
         server, client = socket.socketpair()
         self._client_sockets.add(client)
         self._feeder = LogFeeder(self, server)
+        fileobj = socket.SocketIO(client, 'rb')
+        # Add a "response" object for the streaming client close. We do this
+        # before starting the feeder to avoid races with really fast logs.
+        fileobj._response = self._feeder
```

Comment:
Setting a private attribute on an undocumented class 👏 

...it does clean things up a fair bit, though.